### PR TITLE
Add link attributes by allowing link width and color to be controlled by the user

### LINKED ISSUE #47: Implement link attributes
For now this will likely be restricted to stroke and color

### PR DESCRIPTION
**Pull request by JackWilb**
Closes #47

### CODE REVIEW

File: src/components/NodeLink/functionUpdateVis.js
Comment (waxlamp) [inline src/components/NodeLink/functionUpdateVis.js]: Rather than using an existing variable directly as the link width, it's more likely that we'll want to map a numeric variable to a width using a [d3 continuous scale](https://github.com/d3/d3-scale#continuous-scales). But this is a good start for the infrastructure to handle that.

File: src/components/NodeLink/functionUpdateVis.js
Comment (waxlamp) [inline src/components/NodeLink/functionUpdateVis.js:319]: This is great, though we will also want to support continuous colormaps as well. That can certainly be added in a future PR.

File: src/components/Controls.vue
Comment (waxlamp) [inline src/components/Controls.vue]: It occurs to me that we may also want to allow the user to set width/color of the links based on data residing on the source or target node. That's something to explore in a future PR. But by the same token, I'm not sure it makes sense to offer `table` as an option here, since currently we only allow a single edge table per graph, nor `source` and `target` by themselves, since those turn out to just be IDs for the nodes connected at either end.

**Review by waxlamp — DISMISSED** [review]
This is a great start. I'm good with merging this if we also open issues related to the stuff I brought up in the comments. Let me know what you think.

File: src/components/NodeLink/functionUpdateVis.js
Comment (JackWilb) [reply src/components/NodeLink/functionUpdateVis.js]: Agreed, that's definitely going to be an issue. I just didn't even think of it since I was only using the group var. I think that we should fix this before merging

File: src/components/NodeLink/functionUpdateVis.js
Comment (JackWilb) [reply src/components/NodeLink/functionUpdateVis.js:319]: Opened #57

File: src/components/Controls.vue
Comment (JackWilb) [reply src/components/Controls.vue]: Opened #58 for the first part of your comment. I'll remove table as an option and source and target

Comment (JackWilb) [reply src/components/Controls.vue]: Fixed in 2a77ea5

File: src/components/NodeLink/functionUpdateVis.js
Comment (JackWilb) [reply src/components/NodeLink/functionUpdateVis.js]: Fixed in aff4fb6

File: src/components/NodeLink/NodeLink.vue
Comment (waxlamp) [inline src/components/NodeLink/NodeLink.vue:143]: It would be good to lift the two dependencies out into a destructuring declaration at the top of the function, just to make it visually more apparent what these dependencies are, but that's a minor style improvement that's not blocking this PR.

But hopefully you've gotten a taste for why computed props are nice to work with :cupid:

**Review by waxlamp — APPROVED** [review]
image

**Pull request merged**
